scripts/delete_lambda_versions.py

```diff
@@ -0,0 +1,49 @@
+"""
+Delete all but the latest published version of every AWS Lambda function in the
+current deployment.
+"""
+
+import logging
+
+from azul import (
+    config,
+    reject,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.lambdas import (
+    Lambdas,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    require(config.terraform_component == '',
+            'Cannot be run with a terraform component',
+            config.terraform_component)
+    delete_lambda_versions()
+
+
+def delete_lambda_versions():
+    functions = Lambdas().list_lambdas(config.deployment_stage)
+    for function in functions:
+        log.info('Fetching the published versions of %r', function.name)
+        response = aws.lambda_.list_versions_by_function(FunctionName=function.name)
+        reject('NextMarker' in response, 'Unexpected paged response', response)
+        for version in response['Versions']:
+            version = version['Version']
+            if version != '$LATEST':
+                log.info('Deleting published version %r of %r', version, function.name)
+                aws.lambda_.delete_function(FunctionName=function.name,
+                                            Qualifier=version)
```

Comment:
The bulk of this should live in lambdas.py.

The smell that gives this away is the fact that you use `Lambdas` and `aws.lambda_`. The former is meant as an Azul-specific facade for the latter.